storage/jellyfish-merkle/src/node_type/mod.rs

```diff
@@ -136,28 +136,40 @@ impl NodeKey {
     }
 }
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+pub enum NodeType {
+    Leaf,
+    InternalLegacy,
+}
+
 /// Each child of [`InternalNode`] encapsulates a nibble forking at this node.
 #[derive(Clone, Debug, Eq, PartialEq)]
 #[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
 pub struct Child {
-    // The hash value of this child node.
+    /// The hash value of this child node.
     pub hash: HashValue,
-    // `version`, the `nibble_path` of the ['NodeKey`] of this [`InternalNode`] the child belongs
-    // to and the child's index constitute the [`NodeKey`] to uniquely identify this child node
-    // from the storage. Used by `[`NodeKey::gen_child_node_key`].
+    /// `version`, the `nibble_path` of the ['NodeKey`] of this [`InternalNode`] the child belongs
+    /// to and the child's index constitute the [`NodeKey`] to uniquely identify this child node
+    /// from the storage. Used by `[`NodeKey::gen_child_node_key`].
     pub version: Version,
-    // Whether the child is a leaf node.
-    pub is_leaf: bool,
+    /// Indicates if the child is a leaf, or if it's an internal node, the total number of leaves
+    /// under it
```

Comment:
Feels a bit misleading, since I cannot find the total number of leaves under it from this field.